### PR TITLE
MDEXP-426 - Escape control characters in quoted literals

### PR DESCRIPTION
**Pull request by obozhko-folio**
[MDEXP-426](https://issues.folio.org/browse/MDEXP-426) - Escape control characters in quoted literals

## Purpose
Handle error gracefully if SRS record contains not escaped control characters (for example, \007).

## Approach
- handle json parse unchecked exception when parser encounters with control character
- add test case

## Note
Control characters like \007 cannot be added to SRS record in the database, so that this case should be tested only through the changes directly in the code.

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/service/export/LocalFileSystemExportService.java [hunk 1]
```diff
@@ -94,6 +101,36 @@ public void exportSrsRecord(Pair<List<String>, Integer> marcToExport, ExportPayl
     }
   }
 
+  private void handleControlCharacters(String jsonRecord, String jobExecutionId, OkapiConnectionParams params) {
+    final String controlCharacterMarker = UUID.randomUUID().toString();
+    ErrorCode errorCode = ERROR_MARC_RECORD_CONTAINS_CONTROL_CHARACTERS;
+    // Replace all control characters with markers, otherwise JsonObject cannot be instantiated.
+    jsonRecord = jsonRecord.replaceAll(CONTROL_CHARACTERS_REPLACE_PATTERN, controlCharacterMarker);
```
Comment (alekGbuz) [inline src/main/java/org/folio/service/export/LocalFileSystemExportService.java:108]: Is it possible to use some exists escape utils?

File: src/test/java/org/folio/service/export/ExportServiceUnitTest.java [hunk 2]
```diff
@@ -115,6 +116,23 @@ void shouldSaveErrorLog_whenJsonCannotBeConvertedToMarcRecordBecauseOfLargeSize(
...
...uage\"},{\"v\":\"Readers.\"}],\"ind1\":\" \",\"ind2\":\"0\"}},{\"650\":{\"subfields\":[{\"a\":\"Russian literature\"},{\"y\":\"19th century.\"}],\"ind1\":\" \",\"ind2\":\"0\"}},{\"923\":{\"subfields\":[{\"g\":\"very_long_volume very_long_volume very_long_volume very_long_volume very_long_volume very_long_volume very_long_volume \"},{\"3\":\"ho00000000002\"}],\"ind1\":\" \",\"ind2\":\" \"}},{\"945\":{\"subfields\":[{\"a\":\"491.786\"},{\"b\":\"G5651\"},{\"g\":\"1\"},{\"i\":\"33082017611558\"},{\"j\":\"0\"},{\"l\":\"w3   \"},{\"o\":\"-\"},{\"p\":\"$76.68\"},{\"q\":\"-\"},{\"r\":\"-\"},{\"s\":\"-  \"},{\"t\":\"0\"},{\"u\":\"3\"},{\"v\":\"0\"},{\"w\":\"0\"},{\"x\":\"3\"},{\"y\":\".i21033298\"},{\"z\":\"05-22-15\"}],\"ind1\":\" \",\"ind2\":\" \"}},{\"994\":{\"subfields\":[{\"a\":\"C0\"},{\"b\":\"TXM\"}],\"ind1\":\" \",\"ind2\":\" \"}},{\"998\":{\"subfields\":[{\"a\":\"w\"},{\"b\":\"05-26-15\"},{\"c\":\"m\"},{\"d\":\"a  \"},{\"e\":\"-\"},{\"f\":\"eng\"},{\"g\":\"nyu\"},{\"h\":\"4\"},{\"i\":\"1\"}],\"ind1\":\" \",\"ind2\":\" \"}},{\"999\":{\"subfields\":[{\"s\":\"b26859f1-8c79-47b3-b047-1ecde668492f\"},{\"i\":\"6666df22-5df3-412b-b9cc-cbdddb928e93\"}],\"ind1\":\"f\",\"ind2\":\"f\"}}]}\n";
```
Comment (alekGbuz) [inline src/test/java/org/folio/service/export/ExportServiceUnitTest.java:122]: String record looks a quite large, maybe is it better to read it from some source for tests.

File: src/main/java/org/folio/service/export/LocalFileSystemExportService.java (hunk 1 above)
Comment (obozhko-folio) [reply src/main/java/org/folio/service/export/LocalFileSystemExportService.java:108]: I tried, but utils I could find are not working as expected.

File: src/test/java/org/folio/service/export/ExportServiceUnitTest.java (hunk 2 above)
Comment (obozhko-folio) [reply src/test/java/org/folio/service/export/ExportServiceUnitTest.java:122]: I tried, but in this case all control characters in the obtained string have been already escaped.

**Review by IlliaDaliek — APPROVED** [review]

File: src/test/java/org/folio/service/export/ExportServiceUnitTest.java (hunk 2 above)
Comment (obozhko-folio) [reply src/test/java/org/folio/service/export/ExportServiceUnitTest.java]: I reduced test record.

**Review by alekGbuz — APPROVED** [review]

**Pull request merged**